Fix broken timeline when using [dir="rtl"]

in0ni opened the pull request:
When `dir="rtl"` is set on the `<html>` element, the slidenav (prev/next) and timenav (slider) break. This can be easily replicated when developing locally and simply adding this attribute to the "Bundled TimelineJS test page".

The following fixes are quite minor and non-intrusive:
* The css will only be applied when [dir="rtl"] is present on a parent element
* The slider is set to always use `direction: ltr`
   * Using the above solution is the simplest solution, due to the absolute positioning logic which might bet too complicated if modified for this case
   * I have not found any regressions in the display of slider elements, even when using the `tl-rtl` class (they continue to display properly for rtl languages)

I have added a few less variables, as I found the hard-coded values to be a potential problem (for slide-nav width/margin).

Cheers!

in0ni commented:
I have found another issue with the absolute positioning of `tl-slidenav-content-container` inside the `<button>` element. On sites that display an outline on focus of actionable items (for accessibility) the outline is incorrect -- since the actual dimensions of the button are basically 0.

Is there a reason for this absolute positioning? Just curious, don't want to even attempt if it might be problematic to remove. I have it partially working but have to run.

JoeGermuska commented:
If there's a reason for the absolute positioning, it's lost in time. I'll just note that CSS has advanced a lot since TimelineJS was first created, so there may be a better way to do it now. I mean, I'm not sure today one would even want `div.tl-slidenav-content-container` since it's the only child of the `button`

I wish we had a better way to test for layout regressions besides eyeballing, but generally, if you see a way to make it work without `position: absolute` i'd consider it (admitting that I don't do a very good job of monitoring PR action in this repo)

in0ni commented:
Hey @JoeGermuska,

Agreed, I noticed the css/markup could use some updating/love. It could definitely be simplified, I would love to contribute but time is lacking. Removing the absolute positioning actually simplifies things a bit and allowed me to remove code that is no longer needed. I have tested and all works well:
- Checked landscape/portrait/mobile/skinny
- Checked on my local project with full RTL support (both with `dir="rtl"` and setting in timelinejs3)
- The slider navigation buttons now properly display the focusable area which is good for a11y.

I have removed the `&nbsp;` since when using `[dir="rtl"]` this spacer is placed in the front of the icon and was causing additional space which I was compensating for with odd logic, and now that is no longer needed (see last commit)

Placement of the buttons is now simplified by ensuring the right/left positions are always at 0 -- this avoids a good deal of pixel pushing that was being done.

Lastly, I had to set `direction: ltr` on the previous button... this is a quick-fix as the proper solution sent me down a rabbit hole of multiple changes... which to be honest I rather not go down (has to do with the 100px container width and 80px content width). Would rather not increase the possibility of introducing a regression.

in0ni commented:
Hi @JoeGermuska,

I wonder if this PR would be able to be merged soon. Sorry to apply some pressure, this is more due to a local problem on the project I am working on. Having the source patched in an automated fashion is a bit complicated, and simply using the CDN of the release would be ideal.

If there is anything on my end that could help speed this up, please let me know. I greatly appreciate it in advance.
